### DISCUSSION TITLE
Wasm Platform Conditional

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [unoplatform/uno#12857] [Q&A] [answered]
Hi ! 

I'm trying to get some platform-specific code set up in my Uno Platform project running on Windows & WASM. However, this doesn't seem to be working. Everything compiles correctly and builds, but the text doesn't change on Wasm specifically. I don't know if I'm using the right `#if` conditional or not for Web Assembly - the documentation seems to reference an outdate way of achieving this (splitting partial classes across project heads).

### Markup
```xaml
<UserControl
    x:Class="ConditionalTest"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:HoloManager"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:wasm="http://uno.ui/wasm"
    xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    mc:Ignorable="d wasm"
    d:DesignHeight="300"
    d:DesignWidth="400">

    <Grid>
        <wasm:TextBlock x:Name="WasmTextBlock" Text="Lorem Ipsum" VerticalAlignment="Center"/>
        <win:TextBlock x:Name="WindowsTextBlock" Text="Lorem Ipsum" VerticalAlignment="Center"/>
    </Grid>
</UserControl>
```
### Code-Behind
```csharp
public partial class ConditionalTest : UserControl
{
    public ConditionalTest()
    {
        this.InitializeComponent();
#if !HAS_UNO || HAS_UNO_WASM || __WASM__
        ChangeTextBlock();
#endif
    }
}
```

### Wasm Conditional
```csharp
#if HAS_UNO_WASM || __WASM__
public partial class ConditionalTest : UserControl
{
    private async void ChangeTextBlock()
    {
        WasmTextBlock.Text = "Hello from the Web";
    }        
}
#endif
```
### Windows Conditional
```csharp
#if !HAS_UNO
public partial class ConditionalTest : UserControl
{
    private async void ChangeTextBlock()
    {
        WindowsTextBlock.Text = "Hello from Windows";
    }        
}
#endif
```

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. Please see https://github.com/unoplatform/uno/discussions/12827#discussioncomment-6409429, which contains an answer to your specific question.

**Comment by BertrandDeSaintBezier** [comment]
As suggested in the linked post, the following workaround does the job fine 👌

```csharp
if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
{
    GeneralTextBlock.Text = "Hello from the Web";
}
```
This used in combination with an `else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))` or the `#if !HAS_UNO` pragma pretty much covers the need for WASM/WinUI platform-conditional compilation.

Thanks !
